Bottom: 7ca2cf1eadfad220e47d6b9078cf3d0be56371dd
Top:    07afe03b283a81b9ba61be1da15c9e67dac8dce6
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-06-04 16:34:54 -0500

New role gitlab-user


---
diff --git a/roles/gitlab-user/defaults/main.yml b/roles/gitlab-user/defaults/main.yml
new file mode 100644
index 00000000..fd1bdcdd
--- /dev/null
+++ b/roles/gitlab-user/defaults/main.yml
@@ -0,0 +1,5 @@
+---
+# defaults file for gitlab-user
+
+gitlab_hostname: "127.0.0.1"
+gitlab_uri: "https://{{ gitlab_hostname }}/"
diff --git a/roles/gitlab-user/tasks/main.yml b/roles/gitlab-user/tasks/main.yml
new file mode 100644
index 00000000..5b538c01
--- /dev/null
+++ b/roles/gitlab-user/tasks/main.yml
@@ -0,0 +1,57 @@
+---
+# tasks file for gitlab-user
+
+- assert:
+    that:
+      - new_user is defined
+      - new_user_name is defined
+      - new_user_email is defined
+      - new_user_pass is defined
+
+# Login as root user
+- name: Get access token for admin user
+  uri:
+    url: "{{ gitlab_uri }}/oauth/token"
+    method: POST
+    body_format: json
+    body:
+      grant_type: password
+      username: root
+      password: "{{ gitlab_root_password }}"
+    return_content: yes
+  register: auth
+
+# Search for buildbot user
+- name: Find user
+  uri:
+    url: "{{ gitlab_uri }}/api/v4/users?username={{ new_user }}"
+    method: GET
+    headers:
+      Authorization: "Bearer {{ auth.json.access_token }}"
+    return_content: yes
+  register: founduser
+
+- name: Save new user
+  set_fact:
+    user: '{{ founduser.json[0] }}'
+  when: founduser.json|length > 0
+
+- name: Create missing user
+  block:
+  - uri:
+      url: "{{ gitlab_uri }}/api/v4/users"
+      method: POST
+      headers:
+        Authorization: "Bearer {{ auth.json.access_token }}"
+      return_content: yes
+      body_format: json
+      body:
+        username: "{{ new_user }}"
+        email: "{{ new_user_email }}"
+        password: "{{ new_user_pass }}"
+        name: "{{ new_user_name }}"
+    register: newuser
+
+  - set_fact:
+      user: '{{ newuser.json[0] }}'
+  when: founduser.json|length == 0
diff --git a/roles/gitlab-user/vars/main.yml b/roles/gitlab-user/vars/main.yml
new file mode 100644
index 00000000..612a06bd
--- /dev/null
+++ b/roles/gitlab-user/vars/main.yml
@@ -0,0 +1,2 @@
+---
+# vars file for gitlab-user
\ No newline at end of file
